target/linux/pistachio/base-files/lib/pistachio.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+PISTACHIO_BOARD_NAME=
+PISTACHIO_MODEL=
+
+pistachio_board_detect() {
+	local name=$(grep "^machine" /proc/cpuinfo | sed "s/machine.*: IMG \(.*\) (\(.*\))/\1/g" | awk '{print tolower($1)}')
+	local model=$(grep "^machine" /proc/cpuinfo | sed "s/machine.*: IMG \(.*\) (\(.*\))/\2/g")
```

Comment:
The problem with that is that you would get these values:
```
/tmp/sysinfo/board_name = img,pistachio-marduk
/tmp/sysinfo/model = IMG Marduk (Creator Ci40)
```
hence we currently parse the data. The two solutions I can think of are either:
* move the parsing to a custom 10_sysinfo for marduk and switch to use proc
* patch the dts file and drivers to use expected strings

My preference is the former, I can find one similar thing done [here](https://github.com/lede-project/source/blob/master/target/linux/rb532/base-files/lib/preinit/10_sysinfo).